File: src/main/java/com/me/DenSys/app/Doctor.java
package com.me.DenSys.app;

import javax.persistence.Column;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import java.sql.Date;

public class Doctor {

    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    @Column(name="IIN")
    private Long iIN;
    @Column(name="DateOfBirth")
    private Date DoB;
    @Column(name="FirstName")
    private String firstName;
    @Column(name="Surname")
    private String surname;
    @Column(name="MiddleName")
    private String middleName;
    @Column(name="ContactNumber")
    private String contactNumber;
    @Column(name="DepartmentId")
    private String departmentId;
    @Column(name="SpecializationId")
    private String specializationId;
    @Column(name="Experience")
    private Integer experience;
    @Column(name="Category")
    private Integer price;
    @Column(name="ScheduleDetails")
    private String scheduleDetails;
    @Column(name="Degree")
    private String degree;
    @Column(name="Rating")
    private Double rating;
    @Column(name="Address")
    private String address;
    @Column(name="Url")
    private String url;

}

File: flutter-frontend/lib/domain/models/doctor.dart
import 'package:freezed_annotation/freezed_annotation.dart';

part 'doctor.freezed.dart';
part 'doctor.g.dart';

@freezed
class Doctor with _$Doctor {
  const factory Doctor({
    required DateTime dateOfBirth,
    required String iin,
    required String name,
    required String surname,
    required String middlename,
    required String contactNumber,
    required String departmentId,
    required String specializationDetailsId,
    required String experienceInYears,
    required String doctorCategory,
    required String priceOfAppointment,
    required String degree,
    required String rating,
    required String address,
    required String? homepageUrl,
  }) = _Doctor;

  factory Doctor.fromJson(Map<String, dynamic> json) => _$DoctorFromJson(json);
}
